apps/kg/dataloader/KGDataset.py

```diff
@@ -150,27 +291,201 @@ def read_triple(self, path, mode, skip_first_line=False):
+                tails.append(int(t))
+                rels.append(int(r))
+        heads = np.array(heads, dtype=np.int64)
+        tails = np.array(tails, dtype=np.int64)
+        rels = np.array(rels, dtype=np.int64)
+        print('Finished. Read {} {} triples.'.format(len(heads), mode))
+        return (heads, rels, tails)
 
-def get_dataset(data_path, data_name, format_str):
-    if data_name == 'Freebase':
-        dataset = KGDataset2(data_path, data_name)
-    elif format_str == '1':
-        dataset = KGDataset1(data_path, data_name)
+def get_dataset(data_path, data_name, format_str, files=None):
+    if format_str == 'built_in':
+        if data_name == 'Freebase':
+            dataset = KGDatasetFreebase(data_path)
+        elif data_name == 'FB15k':
+            dataset = KGDatasetFB15k(data_path)
+        elif data_name == 'FB15k-237':
+            dataset = KGDatasetFB15k237(data_path)
+        elif data_name == 'wn18':
+            dataset = KGDatasetWN18(data_path)
+        elif data_name == 'wn18rr':
+            dataset = KGDatasetWN18rr(data_path)
+        else: 
+            assert False, "Unknown dataset {}".format(data_name)
+    elif format_str.startswith('raw_udd'):
+        # user defined dataset
+        format = format_str[8:]
+        dataset = KGDatasetUDDRaw(data_path, data_name, files, format)
+    elif format_str.startswith('udd'):
+        # user defined dataset
+        format = format_str[4:]
+        dataset = KGDatasetUDD(data_path, data_name, files, format)
     else:
-        dataset = KGDataset2(data_path, data_name)
+        assert False, "Unknown format {}".format(format_str)
 
     return dataset
 
 
 def get_partition_dataset(data_path, data_name, format_str, part_id):
     part_name = os.path.join(data_name, 'part_'+str(part_id))
-
-    if data_name == 'Freebase':
-        dataset = KGDataset2(data_path, part_name, read_triple=True, only_train=True)
-    elif format_str == '1':
-        dataset = KGDataset1(data_path, part_name, read_triple=True, only_train=True)
+    if format_str == 'built_in':
+        if data_name == 'Freebase':
+            dataset = KGDatasetFreebase(data_path, part_name, read_triple=True, only_train=True)
+        elif data_name == 'FB15k':
+            dataset = KGDatasetFB15k(data_path, part_name, read_triple=True, only_train=True)
+        elif data_name == 'FB15k-237':
+            dataset = KGDatasetFB15k237(data_path, part_name, read_triple=True, only_train=True)
+        elif data_name == 'wn18':
+            dataset = KGDatasetWN18(data_path, part_name, read_triple=True, only_train=True)
+        elif data_name == 'wn18rr':
+            dataset = KGDatasetWN18rr(data_path, part_name, read_triple=True, only_train=True)
+        else:
+            assert False, "Unknown dataset {}".format(data_name)
+    elif format_str == 'raw_udd':
+        # user defined dataset
+        assert False, "When using partitioned dataset, we assume dataset will not be raw"
```

Comment:
why is it not available yet?

Reply:
For distributed training, we should ask user to provide processed dataset. As each partition may not have the whole entity nodes, generating id mapping is dangerous.